qiskit_optimization/converters/linear_equality_to_penalty.py

```diff
@@ -120,6 +121,9 @@ def convert(self, problem: QuadraticProgram) -> QuadraticProgram:
         else:
             self._dst.maximize(offset, linear, quadratic)
 
+        # Update the penalty to the one just used
+        self._penalty = penalty  # type: float
```

Comment:
@a-matsuo forgot to mention that I am updating `_penalty` in here instead of immediately after calling `_auto_define_penalty()` because I thought it would be better to update it after the conversion was made, so the value is not updated it something failed in the middle, though I can move it to line 64 as suggested :)

Reply:
I see. `penalty` is set on line 125. I think it works better for the case that value is not updated as you mentioned.